src/playlist-maker.js

```diff
@@ -2,6 +2,60 @@ import videojs from 'video.js';
 import playItem from './play-item';
 import * as autoadvance from './auto-advance';
 
+/**
+ * Generate a unique id for each playlist item object. This id will be used to determine
+ * index of an item in the playlist array for cases where there are multiple items with
+ * the same source set.
+ *
+ * @private
+ *
+ * @param {Array}
+ *         An array of playlist items
+ *
+ * @return {Array}
+ *          The list of playlist items with unique ids
+ */
+const generatePlaylistItemId = (arr) => {
+  const list = [];
+
+  for (let i = 0; i < arr.length; i++) {
+    const uuid = Date.now().toString(36) + Math.random().toString(36).substr(2, 5);
+
+    if (typeof arr[i] === 'object') {
+      arr[i].playlistItemId_ = uuid;
+      list.push(arr[i]);
+    }
+  }
+
+  return list;
+};
+
+/**
+ * Look through an array of playlist items for a specific playlist item id.
+ *
+ * @private
+ * @param   {Array}
+ *           An array of playlist items to look through
+ *
+ * @param   {Player}
+ *           The player containing a playlist
+ *
+ * @return  {Number}
+ *           The index of the playlist item or -1 if not found
+ */
+const indexInPlaylistItemIds = (arr, player) => {
```

Comment:
probably better to just pass in the `currentPlaylistItemId_` instead of `player`